src/app/views/query-runner/request/RequestHeaders.tsx

```diff
@@ -0,0 +1,63 @@
+import { IconButton, TextField } from 'office-ui-fabric-react';
+import React from 'react';
+
+interface IRequestHeadersControl {
+    handleOnHeaderDelete: Function;
+    handleOnHeaderNameChange: Function;
+    handleOnHeaderValueChange: Function;
+    handleOnHeaderValueBlur: Function;
+    headers: Array<{ name: string; value: string; }>;
+}
+
+export const RequestHeadersControl = ({
+    handleOnHeaderDelete,
+    handleOnHeaderNameChange,
+    handleOnHeaderValueChange,
+    handleOnHeaderValueBlur,
+    headers,
+}: IRequestHeadersControl) => {
+    const headersList = (
```

Comment:
HeaderList can be it's own component. Making it it's own component will make it easier to read through RequestHeadersControl